source-mysql/discovery.go

```diff
@@ -433,6 +431,20 @@ func (t *mysqlColumnType) translateRecordField(val interface{}) (interface{}, er
 	return val, fmt.Errorf("error translating value of complex column type %q", t.Type)
 }
 
+func (t *mysqlColumnType) encodeKeyFDB(val any) (tuple.TupleElement, error) {
```

Comment:
Do we think this method may eventually allow more types? Since it only allows `enum` here it seems kind of unnecessary vs. handling it in `encodeKeyFDB`, but it would make sense if there was an expectation to extend it in the future.